api/src/services/internal/update_internal_user_role.py

```diff
@@ -0,0 +1,53 @@
+import logging
+
+import src.adapters.db as db
+from src.api.route_utils import raise_flask_error
+from src.constants.lookup_constants import RoleType
+from src.db.models.user_models import InternalUserRole, LinkExternalUser, LinkRoleRoleType, Role
+
+logger = logging.getLogger(__name__)
+
+
+def update_internal_user_role(
+    db_session: db.Session, internal_role_id: str, user_email: str
+) -> None:
+    # Query the role table and verify it is an internal role
+    role = (
+        db_session.query(Role)
+        .join(LinkRoleRoleType, Role.role_id == LinkRoleRoleType.role_id)
+        .filter(Role.role_id == internal_role_id, LinkRoleRoleType.role_type == RoleType.INTERNAL)
+        .first()
+    )
+
+    if not role:
+        raise_flask_error(404, "Internal role not found.")
+
+    # Query user by email using LinkExternalUser
+    external_user = (
+        db_session.query(LinkExternalUser)
+        .filter(LinkExternalUser.email == user_email)
+        .one_or_none()
+    )
+
+    if not external_user:
+        raise_flask_error(404, "User not found.")
+
+    target_user_id = external_user.user_id
+
+    # Check if the user already has this specific role assigned
+    existing_assignment = (
+        db_session.query(InternalUserRole)
+        .filter_by(user_id=target_user_id, role_id=internal_role_id)
+        .first()
+    )
+
+    if existing_assignment:
+        raise_flask_error(422, "User already has this role.")
+
+    # Add the new internal role assignment
+    new_assignment = InternalUserRole(user_id=target_user_id, role_id=internal_role_id)
+    db_session.add(new_assignment)
+
+    db_session.commit()
```

Comment:
You'd probably want to wrap this with a context manager instead of directly committing, something like
`with db.session.begin()` then you don't need the `db_session.commit()`.